packages/generator/src/v1.1/generateActions.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ActionParameterType } from "@osdk/gateway/types";
+import type { MinimalFs } from "../MinimalFs";
+import { getModifiedEntityTypes } from "../shared/getEditedEntities";
+import { formatTs } from "../util/test/formatTs";
+import type { WireOntologyDefinition } from "../WireOntologyDefinition";
+
+export async function generateActions(
+  ontology: WireOntologyDefinition,
+  fs: MinimalFs,
+  outDir: string,
+) {
+  const importedObjects = new Set<string>();
+  let actionSignatures = [];
+  for (const action of ontology.actionTypes) {
+    const entries = Object.entries(action.parameters);
+
+    const modifiedEntityTypes = getModifiedEntityTypes(action);
+    const addedObjects = Array.from(modifiedEntityTypes.addedObjects);
+    const modifiedObjects = Array.from(modifiedEntityTypes.modifiedObjects);
+    addedObjects.forEach(importedObjects.add, importedObjects);
+    modifiedObjects.forEach(importedObjects.add, importedObjects);
+
+    let jsDocBlock = ["/**"];
+    if (action.description) {
+      jsDocBlock.push(`* ${action.description}`);
+    }
+
+    let parameterBlock = "";
+    if (entries.length > 0) {
+      parameterBlock = `params: { \n`;
+      for (
+        const [parameterName, parameterData] of entries
+      ) {
+        parameterBlock += `"${parameterName}"`;
+        parameterBlock += parameterData.required ? ": " : "?: ";
+        const getTypeScriptType = handleDataType(
```

Comment:
`getTypeScriptName` is a little weird naming wise

Reply:
🙈